tools/clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -237,19 +238,22 @@ namespace {
           if (it->first->isValid() && !it->second->IsSystemFile) {
             // If main file, write that to metadata first.
             // Add the rest to filesMap to sort by name.
+            llvm::SmallString<128> NormalizedPath;
+            llvm::sys::path::native(it->first->getName(), NormalizedPath);
             if (CodeGenOpts.MainFileName.compare(it->first->getName()) == 0) {
               assert(!bFoundMainFile && "otherwise, more than one file matches main filename");
-              AddFile(it->first->getName(), it->second->getRawBuffer()->getBuffer());
+              AddFile(NormalizedPath, it->second->getRawBuffer()->getBuffer());
               bFoundMainFile = true;
             } else {
-              filesMap[it->first->getName()] = it->second->getRawBuffer()->getBuffer();
+              filesMap[NormalizedPath] =
+                  it->second->getRawBuffer()->getBuffer();
             }
           }
         }
         assert(bFoundMainFile && "otherwise, no file found matches main filename");
         // Emit the rest of the files in sorted order.
-        for (auto it : filesMap) {
-          AddFile(it.first, it.second);
+        for (auto &it : filesMap) {
```

Comment:
Previously, this iteration was alphabetical due to use of `std::map`.  Since `StringMap` is a hash map, iteration is not going to be alphabetical.  It should be deterministic, but not predictable.  Maybe we could use `std::string` or `SmallString<128>` as the key and std::map to preserve ordering?